lab_5_scrapper/scrapper.py

```diff
@@ -140,10 +290,74 @@ def get_search_urls(self) -> list:
         Returns:
             list: seed_urls param
         """
+        return self.config.get_seed_urls()
 
 
 # 10
 # 4, 6, 8, 10
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler is a child of Crawler class.
+    """
+    def __init__(self, config: Config) -> None:
+        super().__init__(config)
+
+        self.start_url = self.base_url
+
+        self.already_crawled_path = constants.ASSETS_PATH.parent / 'already_crawled_urls.txt'
+        with open(self.already_crawled_path, 'r', encoding='utf-8') as f:
+            self.already_crawled = set(f.readlines())
+
+        self.article_tag = 'ARTICLE URL: '
+        self.urls = [url[len(self.article_tag)+1::]
+                     for url in self.already_crawled
+                     if self.article_tag in url]
+
+    def find_articles(self) -> None:
+        response = make_request(self.start_url, self.config)
+
+        all_urls = [self.start_url]
+        if not response.ok:
+            return
+
+        article_bs = BeautifulSoup(response.text, "lxml")
+
+        all_urls.extend(self.base_url[:-len('/news'):] + link.get('href')
+                        for link in article_bs.find_all(href=True)
+                        if 'http' not in link.get('href'))
```

Comment:
why do not you use extract url from the sequential crawler (recursive crawler's parent)

Reply:
It would only return the article pages, not the other pages we may want to look at while recursively crawling (seed pages mainly)

Reply:
Here I have a question also. Should I collect core pages that are in the HTML, but are not present as hyperlinks on the page? Like files that are .ico  or .css. It takes time to access them to only return with response.ok = False

Reply:
no, ico and css are not pages by definition